File: frontend/src/entities/Location.ts
export interface Location {
    id: number,
    alias: string,
    street: string,
    city: string,
    buildingNumber: string,
}

export function getFormattedLocation(location: Location): string {
    return location.street + " " + location.buildingNumber + ", " + location.city + " (" + location.alias + " )"
}

export function getLabel(location: Location) {
    return location.alias;
}

export function getID(location: Location) {
    return location.id;
}

File: src/main/java/cz/upce/NNPDASEM1/domain/Location.java
package cz.upce.NNPDASEM1.domain;

import cz.upce.NNPDASEM1.domain.user.User;
import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "NN_LOCATIONS")
public class Location {
    @Id
    @Column(name = "LOCATION_ID")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long locationId;

    @Column(nullable = false)
    private String alias;

    @Column(nullable = false)
    private String city;

    @Column(nullable = false)
    private String street;

    @Column(nullable = false)
    private String buildingNumber;

    @ManyToOne
    @JoinColumn(name = "USER_ID", nullable = false)
    private User user;
}
